crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractCustomResourceHandler.java

```diff
@@ -51,51 +51,57 @@ protected AbstractCustomResourceHandler(Resources resources, boolean parallel) {
-    visitors.add(traversedClassesVisitor);
-
-    visitTypeDef(def, visitors);
-
-    addDecorators(config, def, specReplicasPathDetector.getPath(),
-        statusReplicasPathDetector.getPath(), labelSelectorPathDetector.getPath());
-
-    Map<String, Property> additionalPrinterColumns = new HashMap<>(additionalPrinterColumnDetector.getProperties());
-    additionalPrinterColumns.forEach((path, property) -> {
-      Map<String, Object> parameters = property.getAnnotations().stream()
-          .filter(a -> a.getClassRef().getName().equals("PrinterColumn")).map(AnnotationRef::getParameters)
-          .findFirst().orElse(Collections.emptyMap());
-      String type = AbstractJsonSchema.getSchemaTypeFor(property.getTypeRef());
-      String column = (String) parameters.get("name");
-      if (Utils.isNullOrEmpty(column)) {
-        column = property.getName().toUpperCase();
+      List<Visitor<TypeDefBuilder>> visitors = new ArrayList<>();
+      if (config.specClassName().isPresent()) {
+        visitors.add(specReplicasPathDetector);
       }
-      String description = property.getComments().stream().filter(l -> !l.trim().startsWith("@"))
-          .collect(Collectors.joining(" ")).trim();
-      String format = (String) parameters.get("format");
-
-      resources.decorate(
-          getPrinterColumnDecorator(name, version, path, type, column, description, format));
-    });
+      if (config.statusClassName().isPresent()) {
+        visitors.add(statusReplicasPathDetector);
+      }
+      visitors.add(labelSelectorPathDetector);
+      visitors.add(additionalPrinterColumnDetector);
+      visitors.add(traversedClassesVisitor);
+
+      visitTypeDef(def, visitors);
+
+      addDecorators(config, def, specReplicasPathDetector.getPath(),
+          statusReplicasPathDetector.getPath(), labelSelectorPathDetector.getPath());
+
+      Map<String, Property> additionalPrinterColumns = new HashMap<>(additionalPrinterColumnDetector.getProperties());
+      additionalPrinterColumns.forEach((path, property) -> {
+        Map<String, Object> parameters = property.getAnnotations().stream()
+            .filter(a -> a.getClassRef().getName().equals("PrinterColumn")).map(AnnotationRef::getParameters)
+            .findFirst().orElse(Collections.emptyMap());
+        String type = AbstractJsonSchema.getSchemaTypeFor(property.getTypeRef());
+        String column = (String) parameters.get("name");
+        if (Utils.isNullOrEmpty(column)) {
+          column = property.getName().toUpperCase();
+        }
+        String description = property.getComments().stream().filter(l -> !l.trim().startsWith("@"))
+            .collect(Collectors.joining(" ")).trim();
+        String format = (String) parameters.get("format");
+
+        resources.decorate(getPrinterColumnDecorator(name, version, path, type, column, description, format));
+      });
+    } catch (Exception e) {
+      if (e instanceof IllegalArgumentException) {
```

Comment:
Hello,
I am not sure to understand the purpose of it, why making it a special case? are you expecting to catch this somewhere else ?

Reply:
I don't recall exactly, tbh, but I think the idea is that there's already client code that expects an `IllegalArgumentException` to be thrown in some cases here, so just letting this bubble up as expected, yes.

Reply:
Given this context,
`KubernetesClientException.launderThrowable` is what you want for all cases.

In case that's not what you need, why are we using an if-else instead of adding a different catch clause for IllegalArgumentException?